src/line_opacity.jl

```diff
@@ -92,6 +92,21 @@ function scaled_vdW(vdW::Tuple{F, F}, m, T) where F <: AbstractFloat
 end
 
 
+function move_bounds(λs::AbstractRange, lb, ub, λ₀, window_size)
+    len = length(λs)
+    lb = _dont_oob(Int(cld(λ₀ - window_size - λs[1], step(λs)) + 1), len)
+    ub = _dont_oob(Int(fld(λ₀ + window_size - λs[1], step(λs)) + 1), len)
+    lb,ub
+end
+function _dont_oob(n, l)
+    if n < 1
+        1
+    elseif n > l
+        l
+    else
+        n
+    end
+end
```

Comment:
I have a minor suggestion: replace `_dont_oob` with the [`clamp`](https://docs.julialang.org/en/v1/base/math/#Base.Math.clamp) function from the standard library

```suggestion
function move_bounds(λs::AbstractRange, lb, ub, λ₀, window_size)
    len = length(λs)
    lb = clamp(Int(cld(λ₀ - window_size - λs[1], step(λs)) + 1), 1, len)
    ub = clamp(Int(fld(λ₀ + window_size - λs[1], step(λs)) + 1), 1, len)
    lb,ub
end
```